pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractBaseDispatcher.java

```diff
@@ -234,6 +236,12 @@ && trackDelayedDelivery(entry.getLedgerId(), entry.getEntryId(), msgMetadata)) {
         return totalEntries;
     }
 
+    private void doAcknowledgeMessage(List<Position> positions, AckType ackType, Map<String, Long> properties) {
```

Comment:
Maybe we don't need to use `List<Position> positions` here, looks like we don't have the condition to auto-ack with cumulative in the filter.

Reply:
updated. Thanks for your review.